### PR TITLE
Various macOS fixes

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs [hunk 1]
```diff
@@ -24,16 +26,76 @@ public override bool NeedsLayout
...
 
-				if (ShouldInterceptInvalidate)
+				SetLayoutFlags(LayoutFlag.MeasureDirty | LayoutFlag.ArrangeDirty);
+
+				SetSuperviewNeedsLayout();
+			}
+		}
+
+
+		public override void Layout()
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs:46]: You may want to put Layout after `OnInvalidateMeasure` to make the diff easier to read :)

File: src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs [hunk 2]
```diff
@@ -24,16 +26,76 @@ public override bool NeedsLayout
...
+					_inLayoutSubviews = true;
+
+					if (IsMeasureDirty)
+					{
+						// Add back the Margin (which is normally 'outside' the view's bounds) - the layouter will subtract it again
+						var availableSizeWithMargins = Bounds.Size.Add(Margin);
+						XamlMeasure(availableSizeWithMargins);
+					}
+
+					//if (IsArrangeDirty) // commented until the MEASURE_DIRTY_PATH is properly implemented for iOS
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs:61]:
```suggestion
					//if (IsArrangeDirty) // commented until the MEASURE_DIRTY_PATH is properly implemented for macOS
```

**Comment by MartinZikmund** [comment]
Seems the changes somehow caused the app to crash randomly on startup in some cases, need to investigate

**Comment by jeromelaban** [comment]
Let's skip appkit changes for now.

**Pull request closed without merging**